packages/forms/src/UIForm/fieldsets/Array/Array.component.js

```diff
@@ -6,12 +6,16 @@ import defaultTemplates from '../../utils/templates';
 import defaultWidgets from '../../utils/widgets';
 
 function adaptKeyWithIndex(keys, index) {
-	let indexedKeys = keys;
-	const firstIndexPlaceholder = indexedKeys.indexOf('');
-	if (firstIndexPlaceholder >= 0) {
-		indexedKeys = [...keys];
-		indexedKeys[firstIndexPlaceholder] = index;
+	if (!keys) {
+		return;
 	}
+
+	let firstIndexPlaceholder = keys.indexOf('');
```

Comment:
ok i don't understand why you need to add this bit of code

Reply:
The cases : 
- key = ["my", "array", "", "nested"] for nested items fields
- key = ["my", "array"] : this defines array itself. Each item will receive a key ["my", "array", index]
To check that, I spot the first empty string in the key
- find it:  and replace it, it's a nested element
- not found : it's an item key, I add the index after